CRM/Mailing/BAO/Mailing.php

```diff
@@ -925,19 +925,20 @@ public function getTestRecipients($testParams) {
     $senderId = $session->get('userID');
     list($aclJoin, $aclWhere) = CRM_ACL_BAO_ACL::buildAcl($senderId);
 
-    if (array_key_exists($testParams['test_group'], CRM_Core_PseudoConstant::group())) {
-      $contacts = civicrm_api('contact', 'get', array(
-          'version' => 3,
-          'group' => $testParams['test_group'],
-          'return' => 'id',
-          'options' => array(
-            'limit' => 100000000000,
-          ),
-        )
-      );
+    if (array_key_exists('test_group', $testParams)) {
+      if (array_key_exists($testParams['test_group'], CRM_Core_PseudoConstant::group())) {
```

Comment:
Instead of writing this as a parent check, can we append this to the same if condition with `&&` ? Something like -

    if (!empty($testParams['test_group']) && array_key_exists($testParams['test_group'], CRM_Core_PseudoConstant::group())) {